src/ledger/LedgerTxn.h

```diff
@@ -635,13 +635,20 @@ class LedgerTxn : public AbstractLedgerTxn
     std::unique_ptr<Impl> const& getImpl() const;
 
   public:
+    // WARNING: use useTransaction flag with caution. It does not start a SQL
```

Comment:
How big is the performance difference between using a read-only transaction and no transaction at all? I'm asking because I want to make sure that the performance difference justifies the risk (and by association the giant warning).

Reply:
I made the change to set the transaction as read-only and ran a few tests on my machine with Postgres. The results are similar to what I observed comparing this change to master, i.e. no transaction still yields better performance (in the image below yellow is this change, red is read-only transaction)
<img width="1168" alt="Screen Shot 2021-09-07 at 3 55 23 PM" src="https://user-images.githubusercontent.com/9428003/132429662-9ded49fc-249e-4a27-b78a-97be0dba794f.png">

Reply:
Interesting, thanks for testing! Slightly surprising to me but supported by https://stackoverflow.com/questions/33332706/does-postgresql-run-some-performance-optimizations-for-read-only-transactions.